src/Eventso.Subscription/Observing/Batch/BatchEventObserver.cs

```diff
@@ -34,8 +36,12 @@ public sealed class BatchEventObserver<TEvent> : IObserver<TEvent>, IDisposable
             _messageHandlersRegistry = messageHandlersRegistry;
             _skipUnknown = skipUnknown;
 
-            _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(
-                consumer.CancellationToken);
+            _failedBufferProcessor = FailedBufferProcessorResolver.Resolve(
```

Comment:
looks like a thing that has to be injected, but `Buffer<TEvent>.BufferedEvent` is internal while the `BatchEventObserver` constructor is public